packages/mui-joy/src/styles/CssVarsProvider.spec.tsx

```diff
@@ -14,9 +14,13 @@ declare module '@mui/joy/styles' {
     secondary: PaletteRange;
   }
 
+  interface PaletteRange {
+    1000: string;
+  }
+
   interface PaletteRangeOverrides {
-    '100': false;
-    '120': true;
+    100: false;
+    120: true;
```

Comment:
It's still not clear to me if this can really work, for example:

```
plainHoverBg: getCssVarColor(`palette-${color}-100`),
```

This variable will no longer point to a valid value. Should we expect developers to map all variables to point to this value to something new?

Reply:
You will also need to specify `undefined` as a value to the token (`100: undefined`).

That's the expected behavior. The goal is to let developers be able to customize the color range to match their design system needs (the tokens should not be limited to 50-900 range)

Reply:
Those lines only add/remove the theme typings.

Reply:
Coming a bit late to the party, but this was my comment: https://codesandbox.io/s/joy-cra-typescript-forked-2dfxd5?file=/src/App.tsx I know you can make it technically correct, but components will break. We need to make sure we give detailed instructions of what needs to be done after some specific palette value is removed, for e.g. some tokens will need to be redefined.

Reply:
> but components will break

In my point of view, this break is expected because you are changing the token structure. I agree with you that docs should address this clearly and I plan to add a page to the advanced section of Joy UI customization.